bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/StateProfile.java

```diff
@@ -21,46 +17,35 @@
  * @author Simon Kaufmann - initial contribution and API.
  *
  */
+@NonNullByDefault
 public interface StateProfile extends Profile {
```

Comment:
I wonder if it would make sense to make the method names more self-explanatory here.
After all, ALL methods are called from "the outside", so ideally they could be all "on...". We would then only need to add some information about the sender like
`onCommandFromItem`, `onStateUpdateFromItem`, `onCommandFromHandler` and `onStateUpdateFromHandler` or something like that?

Reply:
+1

Profiles (as well as channels) are between Items and ThingHandlers, they don't belong to one of them.

Thereby stateUpdated / onUpdate was confusing to me too.